pandas/core/index.py

```diff
@@ -1585,6 +1602,13 @@ def get_value(self, series, key):
         Fast lookup of value from 1-dimensional ndarray. Only use this if you
         know what you're doing
         """
+
+        # if we have a .values that is an Index
```

Comment:
could fix, but seems much cleaner to simply dispatch. Otherwise have to write lots of cython code. not really worth it.